presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java

```diff
@@ -270,7 +270,12 @@ public int read(int offset, int[] positions, int positionCount)
                 }
                 else {
                     readNestedStreams(offset, positions, positionCount);
-                    outputPositions = nestedOutputPositions;
+                    if (nestedOutputPositions != null) {
```

Comment:
@bhhari 

- `else` block seems unnecessary
- `nestedOutputPositions != null` check raises a question of why would nestedOutputPositions be null; checking nestedOutputPositionCount might be clearer

```
if (nestedOutputPositionCount  > 0) {
    outputPositions = initializeOutputPositions(outputPositions, nestedOutputPositions, positionCount);
}
```

Reply:
yes this is much clear